frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.unit.spec.js

```diff
@@ -1,325 +1,1651 @@
+                  dateStr: "previous hour",
+                  metricValue: 300,
+                }),
+              },
+            },
+          },
+          {
+            description: "should correctly display previous minute comparison",
+            rows: [
+              ["2019-01-02T09:59", 100],
+              ["2019-01-02T10:00", 300],
+            ],
+            dateUnit: "minute",
+            expected: {
+              ...getMetricProperties({
+                dateStr: "Jan 2, 2019, 10:00 AM",
+                metricValue: 300,
+              }),
+              comparison: {
+                ...getComparisonProperties({
+                  changeType: "increase",
+                  comparisonValue: 100,
+                  dateStr: "previous minute",
+                  metricValue: 300,
+                }),
+              },
+            },
+          },
+          {
+            description:
+              "should correctly display N/A if missing previous period",
+            rows: [
+              ["2017-01-01", 100],
+              ["2019-01-01", 300],
+            ],
+            dateUnit: "year",
+            expected: {
+              ...getMetricProperties({
+                dateStr: "2019",
+                metricValue: 300,
+              }),
+              comparison: {
+                ...getComparisonProperties({
+                  changeType: "missing",
+                  dateStr: "previous year",
+                }),
+              },
+            },
+          },
+        ];
+
+        it.each(testCases)("$description", ({ rows, expected, dateUnit }) => {
+          const insights = [{ unit: dateUnit, col: "Count" }];
+          const trend = computeTrend(
+            series({ rows, cols }),
+            insights,
+            settings,
+          );
+          expect(getTrend(trend)).toEqual(expected);
+        });
+
+        describe(`invalid options`, () => {
+          const comparisonType = COMPARISON_TYPES.PREVIOUS_PERIOD;
+          const settings = {
+            "scalar.field": "Count",
+            "scalar.comparisons": { type: comparisonType },
+          };
+
+          const cols = [
+            DateTimeColumn({ name: "Month" }),
+            NumberColumn({ name: "Count" }),
+          ];
+
+          const testCases = [
+            {
+              description: "should handle no dateUnit supplied",
+              rows: [
+                ["2017-01-01", 10],
+                ["2018-01-01", 100],
+                ["2019-01-01", 300],
+              ],
+              dateUnit: null,
+            },
+          ];
+
+          it.each(testCases)("$description", ({ rows, dateUnit }) => {
+            const insights = [{ unit: dateUnit, col: "Count" }];
+
+            expect(() =>
+              computeTrend(series({ rows, cols }), insights, settings),
+            ).toThrow();
+          });
+        });
+      });
+
+      describe(`comparisonType: ${COMPARISON_TYPES.PERIODS_AGO}`, () => {
```

Comment:
@kulyk You'll want to create a new describe section (after this one) for the new comparison type to validate the behaviors.